c/planarityTestGraphFunctionality.c

```diff
@@ -111,3 +212,130 @@ int transformString(graphP theGraph, char *inputStr)
 	
 	return gp_ReadFromString(theGraph, inputStr);
 }
+
+/*
+	TEST ALL GRAPHS IN .G6
+*/
+
+int testAllGraphs(graphP theGraph, char command, char *inputStr, strOrFileP testOutput)
+{
+	int exitCode = OK;
+
+	char *errorStr = NULL;
+
+	graphP copyOfOrigGraph = NULL;
+	int embedFlags = GetEmbedFlags(command);
+	int numGraphsRead = 0, numOK = 0, numNONEMBEDDABLE = 0;
+
+	G6ReadIterator *pG6ReadIterator = NULL;
+	exitCode = allocateG6ReadIterator(&pG6ReadIterator, theGraph);
+
+	if (exitCode != OK)
+	{
+		ErrorMessage("Unable to allocate G6ReadIterator.\n");
+		return exitCode;
+	}
+
+	exitCode = beginG6ReadIterationFromG6String(pG6ReadIterator, inputStr);
+
+	if (exitCode != OK)
+	{
+		ErrorMessage("Unable to begin .g6 read iteration.\n");
+
+		exitCode = freeG6ReadIterator(&pG6ReadIterator);
+
+		if (exitCode != OK)
+			ErrorMessage("Unable to free G6ReadIterator.\n");
+
+		return exitCode;
+	}
+
+	AttachAlgorithm(pG6ReadIterator->currGraph, command);
+
+	copyOfOrigGraph = gp_New();
+	if (copyOfOrigGraph == NULL)
+	{
+		ErrorMessage("Unable to allocate graph to store copy of original graph before embedding.\n");
+		return NOTOK;
+	}
+
+	exitCode = gp_InitGraph(copyOfOrigGraph, pG6ReadIterator->graphOrder);
+	if (exitCode != OK)
+	{
+		ErrorMessage("Unable to initialize graph datastructure to store copy of original graph before embedding.\n");
+		gp_Free(&copyOfOrigGraph);
+		freeG6ReadIterator(&pG6ReadIterator);
```

Comment:
If you decide to keep checking the free iterator result above, then you have to check it here too. I think we're already in a failure state, so best just to free it without checking since we don't really get more info that could help with the earlier failure.